### PR TITLE
Added "Screen Space Refraction" and "Read Normal Map"

### PR DESCRIPTION
**Pull request by SIsilicon**
Screen Space Refraction is like the regular Refraction shader, but more physically accurate. Like Screen Space _Reflection_, it traces the scene's depth buffer to get accurate refractions. It only works in GLES 3 though.

As for the "Read Normal Map" shader, it takes data from normal map, and creates the normal out of it based on the vertex normal, tangent and binormal. The result is in view space.

### CODE REVIEW

File: addons/visual_shader_node_library/shader_nodes/common/read_normalmap.gd [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+func _get_code(input_vars, output_vars, mode, type):
+	if mode != Shader.MODE_SPATIAL or type != VisualShader.TYPE_FRAGMENT:
+		return ""
+	
+	# Default values
+	var normalmap_data = "vec3(0.5, 0.5, 1.0)"
+	var normal = "NORMAL"
+	var tangent = "TANGENT"
+	var binormal = "BINORMAL"
+	var normal_depth = "NORMAL_DEPTH"
```
Comment (Maujoe) [inline addons/visual_shader_node_library/shader_nodes/common/read_normalmap.gd]: I think you mean NORMALMAP_DEPTH instead of NORMAL_DEPTH.

**Review by Maujoe — COMMENTED** [review]
Awesome, looks good. There is only one little error in the normalmap node, would be nice if you could fix this then it can be merged.

And maybe we could catch the low level GLES2 error with [https://docs.godotengine.org/en/3.2/classes/class_os.html#class-os-method-get-current-video-driver](https://docs.godotengine.org/en/3.2/classes/class_os.html#class-os-method-get-current-video-driver) and show an compatibility warning with push_warning(). But this is secondary I can add gles2 warning after merging if you don't want to spend time on that.

File: addons/visual_shader_node_library/shader_nodes/common/read_normalmap.gd (hunk 1 above)
Comment (SIsilicon) [reply addons/visual_shader_node_library/shader_nodes/common/read_normalmap.gd]: Oh! Ok I'll get that fixed. And I'm ok with adding the warning. :)

Comment (SIsilicon) [reply addons/visual_shader_node_library/shader_nodes/common/read_normalmap.gd:79]: Um, where should I put the warning? 😅

**Comment by SIsilicon** [comment]
Ok. We're good now.

**Pull request merged**